### PR TITLE
Backport role and permission docs to release 0.22.0

### PR DESCRIPTION
**Pull request by AlexejPenner**
* Docs regarding roles and permissions

* Apply suggestions from code review

Co-authored-by: Felix Altenberger <felix@zenml.io>

* Added roles and permissions to the toc

Co-authored-by: Felix Altenberger <felix@zenml.io>

## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/starter-guide/collaborate/roles-and-permissions.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+If you want to inspect which roles have been assigned to which users, 
+simply run:
+
+`zenml role assignment list`
+
+## Limitations
+
+* For the time being all roles apply permissions locally. This may change in the 
+future. Feel free to let us know on Slack or through our Roadmap if this is
```
Comment (htahir1) [inline docs/book/starter-guide/collaborate/roles-and-permissions.md:70]: Please link these (Slack) and (Roadmap)

File: docs/book/starter-guide/collaborate/roles-and-permissions.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+important for you.
+
+* Permissions and roles only become relevant if you are running through a
```
Comment (htahir1) [inline docs/book/starter-guide/collaborate/roles-and-permissions.md:73]: Link to deployed ZenML Server?

File: docs/book/starter-guide/collaborate/roles-and-permissions.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+`zenml role list`
+
+## Default Roles
+
+By default, ZenML comes with two roles.
+
+**admin**: The admin role grants all permissions globally.
+
+**guest**: The guest role lets users change their own user data and grants
```
Comment (htahir1) [inline docs/book/starter-guide/collaborate/roles-and-permissions.md:35]: read access or also 'me' access?

**Review by htahir1 — CHANGES_REQUESTED** [review]
left some comments :-)

**Comment by AlexejPenner** [comment]
@htahir1 thanks for your comments, however this is a cherry picked commit on develop as a backport into the release branch. Wouldn't it make sense to make these changes on develop and create a separate backport?

**Comment by fa9r** [comment]
@AlexejPenner you could also make the changes on develop and then backport both changes at the same time by cherry picking twice ;)

**Comment by fa9r** [comment]
but yes, we definitely need the changes on dev first

**Review by fa9r — COMMENTED** [review]
See above, would recommend to first make the changes @htahir1 requested in a separate mini PR to develop, then cherry pick that PR into this one as well so you can backport both

**Review by htahir1 — APPROVED** [review]
Thats great!!

**Pull request merged**